### ISSUE TITLE
[terra-application-utility] Utility Menu accessibility improvements 

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-framework#1589]
# Feature Request

## Description
In the current component `terra-application-utility` used by the older/previous navigation component `terra-application-layout`, the User Settings/Utility menu has specific failings that prevent it's use from being clearly communicated to assistive technology users. 

---

### Part 1: Utility Menu options
The role and state of the User Options button menu cannot be understood by assistive technology. The role is read as button, but opens a menu. Additionally, items in the menu are not operable as a menu. Each item in the menu is read as a button. (WCAG 4.1.2: Name, Role, Value).

### How to Reproduce
- While Voiceover is enabled, tab to the User Options button on the top right of the navigation bar
- Press Return or Space when focused on the User Options tab, and try tabbing through the menu
  - Each menu item will be read as a "button"

### Proposal
- Add `aria-haspopup="menu"` to the UtilityButton so that it indicates it opens a menu
- Add `role="menu"` to the utility menu body section that contains all the menu items
- Change the items in the utility menu from `role="button"` to `role="menuitem"`
  - all except those listed as 'isReadOnly' which will continue to be read as `role="note"`

---

### Part 2: Close button 

In the User Options menu, the name of the Close User Options button cannot be understood by assistive technology users. This element is read as “Close button Settings”. This incorrectly conveys that Settings is the value or state of the Close User Options button (WCAG 4.1.2: Name, Role, Value). 

This reading of Settings comes from the heading title in the given menu. While this button does close the settings menu, the way assistive technology reads this out is confusing for users as it doesn't describe the state of the close button clearly.

### How to Reproduce
- While Voiceover is enabled, tab to the User Options menu and open the menu
- Tab to the Close button
  - Assistive technology will read, "Close, button, {h2 heading title}"

### Proposal
- Remove the `aria-describedby` attribute from the Close button, so that the heading title is not read after and creating unclear context
- Future fix: test incorporating the heading title as part of the Close button label for proper context and order

---

## Additional Context / Screenshots

**Documented criteria:**
HNCONSUMER-2512

**Full Tech Design of related issues compiled here:**
[consumer-framework-tech-designs / spikes / HNCONSUMER-28288: Investigate accessibility issues with navigation.md](https://github.cerner.com/healtheintent/consumer-framework-tech-designs/blob/c709a5bba99a97b0b6f39805e3d2db0fae2f82c7/spikes/HNCONSUMER-28288:%20Investigate%20accessibility%20issues%20with%20navigation.md#user-options-button-isnt-understood-by-assistive-technology)
[consumer-framework-tech-designs / spikes / HNCONSUMER-28291-profile-nav-modal-a11y.md](https://github.cerner.com/healtheintent/consumer-framework-tech-designs/blob/master/spikes/HNCONSUMER-28291-profile-nav-modal-a11y.md)

## @ Mentions

Will be contribution from Consumer development team
